### PR TITLE
overlapping modules page integration

### PR DESCRIPTION
**Pull request by a0129998**
1. implemented overlappingModules page
2. updated button style

### CODE REVIEW

**Comment by tgqiang** [comment]
From <code>test_mod_taken_together_queries.py</code>:
2 test cases failed (<code>test_query_module_taken_together_entire_list</code> and <code>test_query_module_taken_together_specified_mod</code>)

@nlzz22 Please check backend and/or test cases.

@a0129998 UI issues:
1) Please fix HTML title spelling to the proper case
2) Same goes for the header in the HTML (the <code>h1</code> header)
3) (**Optional**) Can consider wrapping the form for specifying target module with a solid border.

**Comment by nlzz22** [comment]
@tgqiang did u reset the database before you run the tests ? Because I have populated extra information to the database which is essential to the testing of the app.

**Comment by a0129998** [comment]
@tgqiang  changes updated
image
image

**Comment by a0129998** [comment]
also, all test cases passed on my pc on local host

File: components/handlers/index.py [hunk 1]
```diff
@@ -9,13 +9,6 @@
 
 
 class Index(object):
```
Comment (nlzz22) [inline components/handlers/index.py]: Missing class docstring for class Index.

File: test/test_app.py [hunk 2]
```diff
@@ -12,45 +12,6 @@ class TestCode(object):
         This class runs the test cases to test app home page
```
Comment (nlzz22) [inline test/test_app.py]: This file is not linted. Please pylint it thanks.

File: components/handlers/index.py (hunk 1 above)
Comment (a0129998) [inline components/handlers/index.py]: added docstring for class

File: test/test_app.py (hunk 2 above)
Comment (a0129998) [inline test/test_app.py]: linted file

**Comment by a0129998** [comment]
applied changes requested, and removed non-functional links from index

**Comment by tgqiang** [comment]
@xaterz 
The search bar belonging to the <code>DataTable</code> is part of that table. Points 1 & 2 can be addressed as you have mentioned.

For point 3, the display of top 10 results for commonly-taken modules belong to the Individual Module View for module **M** (but this is not implemented yet). The table for Overlapping Modules will display the full result set.

For point 4, aggregation of data, for both semesters in an AY, into a single count may not be reflective of semester-specific data analysis. We could probably display both types of data, but the implementation for overlapping modules should probably be discussed before a decision is made.

Summary: Points 1 to 3 can be easily addressed; Point 4 requires discussion before any change is effected.

**Pull request merged**